### PR TITLE
Add groups DTD and classes.

### PR DESCRIPTION
**Pull request by aptls**
Add the groups feed DTD, classes, and example. The groups DTD
includes a "principal" element which is different from the element
with the same name in the GSA feed DTD, so the groups classes are
placed in a different package.

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -42,6 +42,17 @@
       <arg value="-no-header"/>
       <arg value="examples/gsafeed.dtd"/>
     </exec>
+    <exec executable="xjc">
```
Comment (PapaJoltOmega) [inline build.xml:45]: We are submitting generated classes too, yes? When are the classes generated?  Or is this a reference for posterity?

File: examples/groups.dtd [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (wiarlawd) [inline examples/groups.dtd]: Internally this is groupsfeed.dtd. It looks like the name was changed elsewhere during development, and xmlgroups.dtd works for me, if we're willing to rename the internal file eventually.

File: build.xml (hunk 1 above)
Comment (wiarlawd) [reply build.xml:45]: Gennis included this target in the first CL. It both documents what we used to generate the classes and could be used to generate them again for comparison with the code we're planning to modify. If we end up not using this, or the diffs become unusable, we can just document this somewhere in a README or something.

**Review by wiarlawd — APPROVED** [review]
LGTM++, but we need to settle on a name for the DTD.

**Review by PapaJoltOmega — APPROVED** [review]

**Comment by wiarlawd** [comment]
Let's just rename the DTD to groupsfeed.dtd [sic] to match the existing internal name, and submit this.

**Pull request merged**